.github/workflows/ci_suite.yml

```diff
@@ -51,23 +51,49 @@ jobs:
-          bash tools/ci/check_grep.sh
-          bash tools/ci/check_misc.sh
-          tools/bootstrap/python tools/ticked_file_enforcement/ticked_file_enforcement.py < tools/ticked_file_enforcement/schemas/tgstation_dme.json
-          tools/bootstrap/python tools/ticked_file_enforcement/ticked_file_enforcement.py < tools/ticked_file_enforcement/schemas/unit_tests.json
-          tools/bootstrap/python -m tools.maplint.source --github
-          tools/build/build --ci lint tgui-test
-          tools/bootstrap/python -m define_sanity.check
-          tools/bootstrap/python -m dmi.test
-          tools/bootstrap/python -m mapmerge2.dmm_test
-          ~/dreamchecker > ${GITHUB_WORKSPACE}/output-annotations.txt 2>&1
+      - name: Give Linters A Go
+        id: linter-setup
+        run: ':'
+      - name: Check File Directories
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: bash tools/ci/check_filedirs.sh tgstation.dme
+      - name: Check Changelogs
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: bash tools/ci/check_changelogs.sh
+      - name: Run Grep Checks
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: bash tools/ci/check_grep.sh
+      - name: Check Miscellaneous Files
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: bash tools/ci/check_misc.sh
+      - name: Ticked File Enforcement (tgstation.dme)
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/bootstrap/python tools/ticked_file_enforcement/ticked_file_enforcement.py < tools/ticked_file_enforcement/schemas/tgstation_dme.json
+      - name: Ticked File Enforcement (unit_tests.json)
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/bootstrap/python tools/ticked_file_enforcement/ticked_file_enforcement.py < tools/ticked_file_enforcement/schemas/unit_tests.json
+      - name: Run Map Linter
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/bootstrap/python -m tools.maplint.source
+      - name: Run TGUI Tests
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/build/build --ci lint tgui-test
+      - name: Run Define Sanity Check
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/bootstrap/python -m define_sanity.check
+      - name: Run DMI Tests
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/bootstrap/python -m dmi.test
+      - name: Run Mapmerge2 Tests
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        run: tools/bootstrap/python -m mapmerge2.dmm_test
+      - name: Run DreamChecker
+        if: steps.linter-setup.conclusion == 'success' && !cancelled()
+        id: dreamchecker
+        run: ~/dreamchecker > ${GITHUB_WORKSPACE}/output-annotations.txt 2>&1
```

Comment:
i think the old order of the CI run was such that it would just look nicer without any real thought put into what should go where i think it should go

- All DM Checks (Greps, Ticked File Enforcement, Define Sanity, DreamChecker), since this is what will most often fail and we want to bounce out of doing the stuff with the runners as fast as possible if it's going to fail regardless. We don't need DreamChecker to run last anymore since we use the python script that you wrote. it's also nicer for the greps to run first because then we can cache the `ripgrep` installation a bit faster whenever that clears or for people testing CI on their downstream.
- Other miscellaneous code checks (JSON validifier, fire directories, changelogs, whatever 'misc' is), since it tends to be rather fast.
- The TGUI Checks
- The Icon/DMI Checks
- The Maplint Checks

TGUI could probably also go last, but I'm just fielding this out there since we might as well get a solid structure down now that we have the opportunity to do so.

Reply:
> since this is what will most often fail and we want to bounce out of doing the stuff with the runners as fast as possible if it's going to fail regardless.

Should I ditch the idea of continuing linters even if one step failed then, or should I combine those steps to one?

Reply:
ehhhhhh now that i think about it I think it should still continue rather than fail fast because having to do multiple commits to keep re-running CI is quite a bit of a pain and will probably end up consuming more runners than just doing it all at once. the order should still hold true because having a flagged :x: be thrown up ASAP for stuff like failing DM code is rather useful since it'll show up on the commit in the PR, the PR in the repo, as well as show up in github desktop and the like.